commit c66b0b32fba175d5f096c944d8ec8f9f06299f4a
Author: Daniel Stenberg <daniel@haxx.se>
Date:   Sun Apr 10 19:14:22 2011 +0200

    OpenSSL: no-sslv2 aware
    
    Allow openSSL without SSL2 to be used. This fix is inspired by the fix
    provided by Cristian Rodríguez.
    
    Reported by: Cristian Rodríguez

diff --git a/lib/ssluse.c b/lib/ssluse.c
index 654ffaa..caffdad 100644
--- a/lib/ssluse.c
+++ b/lib/ssluse.c
@@ -1327,8 +1327,13 @@ ossl_connect_step1(struct connectdata *conn,
     req_method = TLSv1_client_method();
     break;
   case CURL_SSLVERSION_SSLv2:
+#ifdef OPENSSL_NO_SSL2
+    failf(data, "OpenSSL was built without SSLv2 support");
+    return CURLE_UNSUPPORTED_PROTOCOL /* CURLE_NOT_BUILT_IN not defined in 7.19.0 */;
+#else
     req_method = SSLv2_client_method();
     break;
+#endif
   case CURL_SSLVERSION_SSLv3:
     req_method = SSLv3_client_method();
     break;
